server/vulnerabilities/goval_dictionary/database.go

```diff
@@ -0,0 +1,71 @@
+package goval_dictionary
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/utils"
+	"strings"
+)
+
+type VulnDataSource interface {
+	// Eval evaluates the current OVAL definition against an OS version and a list of installed software, returns all software
+	// vulnerabilities found.
+	Eval(fleet.OSVersion, []fleet.Software) ([]fleet.SoftwareVulnerability, error)
+}
+
+func NewDB(db *sql.DB) *Database {
+	return &Database{sqlite: db}
+}
+
+type Database struct {
+	sqlite *sql.DB
+}
+
+// Eval evaluates the current goval_dictionary database against an OS version and a list of installed software,
+// returns all software vulnerabilities found.
+func (db Database) Eval(version fleet.OSVersion, software []fleet.Software) ([]fleet.SoftwareVulnerability, error) {
+	searchStmt := `SELECT packages.version, cves.cve_id 
+		FROM packages join definitions on definitions.id = packages.definition_id
+		JOIN advisories ON advisories.definition_id = definitions.id JOIN cves ON cves.advisory_id = advisories.id
+		WHERE packages.name = ? AND packages.arch = ?`
+	vulnerabilities := make([]fleet.SoftwareVulnerability, 0)
+
+	for _, swItem := range software {
+		affectedSoftwareRows, err := db.sqlite.Query(searchStmt, swItem.Name, swItem.Arch)
+		if errors.Is(err, sql.ErrNoRows) {
```

Comment:
Only QueryRow() can return ErrNoRows.  If there are no rows, `affectedSoftwareRows == nil`

Reply:
Cool. Moved that to the appropriate place.